fastapi/dependencies/utils.py

```diff
@@ -86,6 +85,138 @@
+            self.call_type = CallType.ASYNC
+        elif getattr(call, '__fastapi_sync_dependency__', False):
+            self.call_type = CallType.SYNC
+        else:
+            self.call_type = CallType.THREADPOOL
+
+    def copy(self):
+        return DependencyNode(
+            self.call,
+            arguments={},
+            refcount=self.refcount,
+            backref=self.backref,
+            result=None,
+            _copy_args=(
+                self.call_type,
+            )
+        )
+
+    async def task(
+        self,
+        nodes: List['DependencyNode'],
+        request: Request,
+        task_group: Optional[anyio.abc.TaskGroup],
+    ):
+        call = self.call
+        arguments = self.arguments
+        if self.call_type == CallType.GENERATOR:
+            stack = request.scope.get("fastapi_astack")
+            assert isinstance(stack, AsyncExitStack)
+            self.result = await solve_generator(call=call, stack=stack, arguments=arguments)
+        elif self.call_type == CallType.ASYNC:
+            self.result = await call(**arguments)
+        elif self.call_type == CallType.THREADPOOL:
+            self.result = await run_in_threadpool(call, **arguments)
+        else:
+            self.result = call(**arguments)
+
+        for parent_node_idx, argument_name in self.backref:
+            parent_node = nodes[parent_node_idx]
+            if argument_name is not None:
+                parent_node.arguments[argument_name] = self.result
+
+            parent_node.refcount -= 1
+            if parent_node.refcount != 0:
+                continue
+
+            if task_group is not None:
+                task_group.start_soon(parent_node.task, nodes, request, task_group)
+            else:
+                await parent_node.task(nodes, request, task_group)
+
+
+class SolvingPlan(object):
+
+    def __init__(
+        self,
+        nodes,
+        path_params_list,
+        query_params_list,
+        header_params_list,
+        cookie_params_list,
+        background_tasks_param_list,
+        http_connection_param_list,
+        request_param_list,
+        websocket_param_list,
+        response_param_list,
+        security_scopes_param_list,
+        body_params_list,
+    ):
+        self.nodes = nodes
+        self.path_params_list = path_params_list
+        self.query_params_list = query_params_list
+        self.header_params_list = header_params_list
+        self.cookie_params_list = cookie_params_list
+        self.background_tasks_param_list = background_tasks_param_list
+        self.http_connection_param_list = http_connection_param_list
+        self.request_param_list = request_param_list
+        self.websocket_param_list = websocket_param_list
+        self.response_param_list = response_param_list
+        self.security_scopes_param_list = security_scopes_param_list
+        self.body_params_list = body_params_list
+
+        self.sync_tasks_count = sum(
+            1 for node in nodes
```

Comment:
Use variable binding to make 1 clear what it is.

Reply:
Can you give example?
1 is just 1, I need to count nodes with some condition.

Reply:
Then I suggest filtering sync_call_type_node and passing its length to the function.

Reply:
Such technique will create unneeded array:
```
>>> import resource
>>> val = len([1 for i in range(int(1e8)) if i % 2 == 0])
>>> resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
401112
```
vs
```
>>> import resource
>>> val = sum(1 for i in range(int(1e8)) if i % 2 == 0)
>>> resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
10280
```

Reply:
Okay.